src/api/apiRoutes.js

```diff
@@ -0,0 +1,47 @@
+import express from "express"
+import { getGames } from "./helpers.js"
+
+const router = express.Router()
+
+router.get("/get-top-products", (req, res) => {
+  const games = getGames()
+  const gamesSortedByDate = games
+    .sort((a, b) => a.releaseDate - b.releaseDate)
+    .slice(0, 3)
+  return res.send(gamesSortedByDate)
+})
+
+// GAME PRODUCTS LIST
+router.get(`/products`, (req, res) => {
+  const listOfGames = getGames()
+  let searchQuery = req.query.search?.toLowerCase()
+  const categoryQuery = req.query.category
+  const filteredByCategory = listOfGames.filter((game) =>
+    game.category.includes(categoryQuery)
+  )
+  if (!searchQuery && !categoryQuery) {
```

Comment:
Why do you check query parameters right after filtering? Definitely need to do it before filtering

Reply:
I use this filterArray only in places where i definitely know i pass certain condition. Following checking params does not affect this array by any means